src/shaders/line.fragment.glsl

```diff
@@ -22,6 +26,10 @@ void main() {
     float blur2 = (blur + 1.0 / u_device_pixel_ratio) * v_gamma_scale;
     float alpha = clamp(min(dist - (v_width2.t - blur2), v_width2.s - dist) / blur2, 0.0, 1.0);
 
+#if defined( FOG ) && !defined( RENDER_TO_TEXTURE )
+    color = fog_apply_premultiplied(color, v_fog_pos);
```

Comment:
I think you'd have to create a new variable here (out_color to be consistent?), color may be a varying which can't be an l-value (e.g. re-assigned).

It can be defined as a varying in the #pragma expansion here: https://github.com/mapbox/mapbox-gl-js/pull/10496/files#diff-f98842bfa88df3ed6d33b46a71900e77e279166871bb4c77f7cd6eb05e3dc89eR16